misk-hibernate/src/main/kotlin/misk/hibernate/TransformedColumnType.kt

```diff
@@ -0,0 +1,122 @@
+package misk.hibernate
+
+import org.hibernate.HibernateException
+import org.hibernate.engine.spi.SharedSessionContractImplementor
+import org.hibernate.type.spi.TypeConfiguration
+import org.hibernate.type.spi.TypeConfigurationAware
+import org.hibernate.usertype.ParameterizedType
+import org.hibernate.usertype.UserType
+import java.io.Serializable
+import java.sql.PreparedStatement
+import java.sql.ResultSet
+import java.sql.Types
+import java.util.Properties
+import kotlin.reflect.KClass
+import kotlin.reflect.full.primaryConstructor
+
+internal class TransformedColumnType : UserType, ParameterizedType, TypeConfigurationAware {
+  companion object {
+    val TRANSFORMER_CLASS = "transformerClass"
+    val TABLE_NAME = "tableName"
+    val COLUMN_NAME = "columnName"
+    val FIELD = "field"
+    val TARGET_TYPE = "targetType"
+    val ARGUMENTS = "arguments"
+  }
+
+  lateinit var retClass: KClass<*>
+
+  lateinit var transformer: Transformer
+
+  lateinit var typeConfig: TypeConfiguration
+
+  override fun hashCode(x: Any) = x.hashCode()
+
+  override fun deepCopy(value: Any?) = value
+
+  override fun replace(original: Any?, target: Any?, owner: Any?) = original
+
+  override fun equals(x: Any?, y: Any?) = x == y
+
+  override fun returnedClass(): Class<*> = retClass.java
+
+  override fun assemble(cached: Serializable, owner: Any): Any = transformer.assemble(owner, cached)
+
+  override fun disassemble(value: Any): Serializable = transformer.disassemble(value)
+
+  override fun nullSafeSet(st: PreparedStatement, value: Any?, index: Int, session: SharedSessionContractImplementor?) {
+    if (value == null) {
+      // TODO: Replace this with actual class
```

Comment:
Blocking TODO?

Reply:
That's actually a stale comment (the actual class is used). Just pushed to remove.